### PR TITLE
[image_picker] Adds comment for the limit parameter

### PR DESCRIPTION
**Pull request by LinXunFeng**
Fixes https://github.com/flutter/flutter/issues/147773

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/image_picker/image_picker/lib/image_picker.dart
Comment (tarrinneal) [inline packages/image_picker/image_picker/lib/image_picker.dart:116]: There should be `[]` around `ImagePickerAndroid`, and ` `` ` around `useAndroidPhotoPicker`

This also seems a little verbose, maybe shorten it to 

```On Android, [ImagePickerAndroid.useAndroidPhotoPicker] must be set to `true` to use the `limit` functionality```

I'm not sure the `see: [link]` part is necessary either.

Same for the other instance of this comment.

Comment (LinXunFeng) [reply packages/image_picker/image_picker/lib/image_picker.dart]: Thanks for the review, I've readjusted it.

**Comment by tarrinneal** [comment]
I guess, before this gets merged, is this a feature that isn't available in the android sdk? or is it just not implemented properly in the plugin?

**Comment by LinXunFeng** [comment]
I made a reply here(https://github.com/flutter/flutter/issues/147773#issuecomment-2099599805) and adjusted the comment. Please review again.

File: packages/image_picker/image_picker/lib/image_picker.dart
Comment (stuartmorgan) [inline packages/image_picker/image_picker/lib/image_picker.dart:119]: We do not generally get this specific about platform-specific details in the app-facing package:
- They tend to get outdated because they are about details that are controlled in another package.
- They don't scale well to many platforms (e.g., you haven't discussed desktop here, and specifically calling out only Android makes it seem like desktop should be fine, which isn't the case).
- They can't scale to third-party implementations.

I would much prefer we just use [the same comment that we use for the platform interface layer](https://github.com/flutter/packages/blob/822d8852e7003e2366458873b4ecac23ce7f812b/packages/image_picker/image_picker_platform_interface/lib/src/types/multi_image_picker_options.dart#L31).

This level of platform-specific detail is something we could put in the Android package's README.

Comment (LinXunFeng) [reply packages/image_picker/image_picker/lib/image_picker.dart]: Got it.

**Comment by reidbaker** [comment]
It looks like there are some comments here otherwise the PR looks good. Do you plan to update this pr? if not we understand and thank you for the effort you have already spent.

**Review by reidbaker — APPROVED** [review]

**Comment by LinXunFeng** [comment]
@reidbaker, thanks for your review.

I have previously moved the comments about the Android part to the README of `image_picker_android` according to @stuartmorgan 's suggestion. Then the PR is waiting for further review and there is no other update.

Now I have rebased the latest code and resolved the conflict.

File: packages/image_picker/image_picker_android/README.md
Comment (stuartmorgan) [inline packages/image_picker/image_picker_android/README.md:36]: Nit: `[ImagePickerAndroid.useAndroidPhotoPicker]` shouldn't have brackets, since this isn't a Dartdoc comment.

File: packages/image_picker/image_picker_android/CHANGELOG.md
Comment (stuartmorgan) [inline packages/image_picker/image_picker_android/CHANGELOG.md:1]: 0.8.12+3

**Review by stuartmorgan — APPROVED** [review]
LGTM with two small fixes.

File: packages/image_picker/image_picker_android/README.md
Comment (LinXunFeng) [reply packages/image_picker/image_picker_android/README.md]: Thx.

**Review by reidbaker — APPROVED** [review]

**Pull request merged**
